spec/02-integration/03-db/14-dao_spec.lua

```diff
@@ -185,6 +187,33 @@ for _, strategy in helpers.all_strategies() do
       assert.same(new_plugin_config.config.redis.host, read_plugin.config.redis.host)
       assert.same(new_plugin_config.config.redis.host, read_plugin.config.redis_host) -- legacy field is included
     end)
+
+    it("keyauth_credentials can be deleted or selected before run ttl cleanup in background timer", function()
+      local original_keyauth_credentials = bp.keyauth_credentials:insert({
+        consumer = { id = consumer.id },
+      }, { ttl = 5 })
+
+      -- wait for 5 seconds.
+      ngx.sleep(5)
+
+      -- select or delete keyauth_credentials after ttl expired.
+
+      local expired_keyauth_credentials = kong.db.keyauth_credentials:select({ consumer = { id = consumer.id } })
+      assert.is_nil(expired_keyauth_credentials)
+      kong.db.keyauth_credentials:delete({ consumer = { id = consumer.id } })
```

Comment:
maybe we need to add some `wait_until` check after `sleep(5)` to make sure the inserted key has expired,  like this pr: https://github.com/Kong/kong/pull/13235/files. Because sleep(5) and ttl=5 are too close.

we could select the inserted key without `skip_ttl` flag to check if it expired.

```
        helpers.wait_until(function()
          return assert(kong.db.keyauth_credentials:select({ consumer = { id = consumer.id } })
                                  == nil)
        end, 1)
)
```